gspread/worksheet.py

```diff
@@ -2964,3 +2966,26 @@ def cut_range(
         }
 
         return self.spreadsheet.batch_update(body)
+
+    def column_count(self):
+        """Full English alias for .col_count"""
+        return self.col_count()
+
+    def list_protected_ranges(self):
+        """List protected ranges in current Worksheet"""
+        return self.spreadsheet.list_protected_ranges(self.id)
+
+    def protect(self):
+        """Protect all ranges in current Worksheet"""
+        email = get_email_from_somewhere_tbd()  # TODO ?
+        last_cell = rowcol_to_a1(self.row_count, self.col_count)
+        self.add_protected_range(
+            f"A1:{last_cell}",
+            email,
+            description="LOCKED by gspread user",
+        )
```

Comment:
Did we try to call the API to add a new protected range but with no actual range. Something like, instead of giving: `sheet1!A1:Z99` what if we give it: `sheet1` as "range" does the API understand that we want to cover everything and then protects everything?? 🙃